backend/src/yeomanui.ts

```diff
@@ -377,9 +400,22 @@ export class YeomanUI {
    * Also functions cannot be evaluated on client)
    */
   private normalizeFunctions(questions: Environment.Adapter.Questions<any>): Environment.Adapter.Questions<any> {
+    this.addCustomQuestionEventHandlers(questions);
     return JSON.parse(JSON.stringify(questions, YeomanUI.funcReplacer));
   }
 
+  private addCustomQuestionEventHandlers(questions: Environment.Adapter.Questions<any>): void {
+    for (const index in questions) {
+      const question = (questions as any[])[Number.parseInt(index)];
+      const questionHandlers = this.customQuestionEventHandlers.get((question as any)["guiType"]);
```

Comment:
isn't question can have only 1 guiType?

Reply:
Yes, only 1 guiType, but here we're looping through all questions of a particular prompt.

Then, each question can possibly have several event handlers. For example, login and logout for a custom login question type.